src/uu/cp/src/platform/macos.rs

```diff
@@ -63,8 +63,23 @@ pub(crate) fn copy_on_write(
             {
                 // clonefile(2) fails if the destination exists.  Remove it and try again.  Do not
                 // bother to check if removal worked because we're going to try to clone again.
-                let _ = fs::remove_file(dest);
-                error = pfn(src.as_ptr(), dst.as_ptr(), 0);
+                // first lets make sure the dest file is not read only
+                match fs::metadata(dest) {
+                    Ok(md) => {
+                        if md.permissions().readonly() {
+                            return Err(format!(
+                                "failed to clone {source:?} from {dest:?}: permission denied"
```

Comment:
I wonder if there's a better way of doing this, because the error is not _necessarily_ permission denied. Can we get the same "permission denied" by just letting the fallback copy handle this?

To be more specific, can we get the same behaviour by just doing this?
```rust
if fs::metadata(dest).map_or(false |md| !md.permissions().readonly()) {
    // remove and copy again
} 
```

Reply:
Good catch, I pushed a commit that simplifies the permission check, tests seem to be passing